src/mesh/boundary_standard.cxx

```diff
@@ -2820,7 +2962,7 @@ void BoundaryFree_O2::apply(Field3D &f) {
   // Check for staggered grids
   
   CELL_LOC loc = f.getLocation();
-  if(mesh->StaggerGrids && loc != CELL_CENTRE) {
+  if(mesh->StaggerGrids and (loc == CELL_XLOW or loc == CELL_YLOW)) {
```

Comment:
Is `CELL_ZLOW` incompatible with this boundary condition?

Reply:
No, the reason for this condition is that for boundary conditions that don't use a value calculated from expressions, the 'unstaggered' version works for both `CELL_CENTRE` and `CELL_ZLOW`, so I wanted to drop into the `else` case.